benchexec/container.py

```diff
@@ -722,6 +740,37 @@ def escape(s):
     )
 
 
+def make_fuse_overlay_mount(exe, mount, lower, upper, work):
+    logging.debug(
+        "Creating overlay mount: target=%s, lower=%s, upper=%s, work=%s",
+        mount,
+        lower,
+        upper,
+        work,
+    )
+
+    def escape(s):
+        return s.replace(b"\\", rb"\\").replace(b":", rb"\:").replace(b",", rb"\,")
+
+    cmd = (
+        exe,
+        b"-o",
+        b"lowerdir="
+        + escape(lower)
+        + b",upperdir="
+        + escape(upper)
+        + b",workdir="
+        + escape(work),
+        escape(mount),
+    )
+
+    try:
+        subprocess.run(args=cmd, check=True)
```

Comment:
Are there stdout or stderr message that could be written by `fuse-overlayfs`? Then we should maybe collect and log them at debug log level?

Reply:
If fuse-overlayfs runs normally, it will not display any messages (unless the `--debug` option is used). I've used `try-except` to catch `CalledProcessError` when running `fuse-overlayfs`. Is this approach standard? Perhaps we could reconsider the error messages output here.

Reply:
Yes, what you are doing is standard. One could also avoid `check=True`, remove the `try`-`catch` and simply check the exit code of the process. But that does not matter much.

What I would suggest is `stdout=subprocess.PIPE, stderr=subprocess.STDOUT`. And then, if there is an error, add `stdout` (which includes `stderr`) of `fuse-overlayfs` to the error message of BenchExec. And if there is no error but still non-empty output, log the output with level debug.